lib/active_merchant/billing/gateways/quickpay/quickpay_v10.rb

```diff
@@ -15,7 +15,7 @@ def initialize(options = {})
       end
 
       def purchase(money, credit_card_or_reference, options = {})
-        MultiResponse.run(true) do |r|
+        process = MultiResponse.run(true) do |r|
```

Comment:
I should also note, if you look in the Stripe adapter you'll see that it's just `MultiResponse.run do |r|` and since the initializer defaults to false, it'll use the last response.